libpod/container_internal_linux.go

```diff
@@ -1458,11 +1458,35 @@ func (c *Container) generateCurrentUserPasswdEntry() (string, error) {
 	if uid == 0 {
 		return "", nil
 	}
+
 	u, err := user.LookupId(strconv.Itoa(rootless.GetRootlessUID()))
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to get current user")
 	}
-	return fmt.Sprintf("%s:x:%s:%s:%s:%s:/bin/sh\n", u.Username, u.Uid, u.Gid, u.Username, c.WorkingDir()), nil
+
+	// Lookup the user to see if it exists in the container image.
+	_, err = lookup.GetUser(c.state.Mountpoint, u.Username)
+	if err != User.ErrNoPasswdEntries {
+		return "", err
+	}
+
+	// If the user's actual home directory exists, or was mounted in - use
+	// that.
```

Comment:
This is interesting, but somewhat out of scope.

- Is it _safe_? I can easily imagine containers that assume that `~` == `WORKDIR`.
- Also, is it OK to give the container this information about the outside? There is the precedent of telling the container the user name already, sure.

Reply:
The primary case I was trying to capture here was pet containers (Toolbox) which explicitly bind-mount in the user's `/home`. I think we're safe to set in that case, given the user's /home was mounted in explicitly at container creation.

Dropping the case where we set `/home` if it exists in the container rootfs would simplify things and prevent the leakage when the home directory was not explicitly added, so I think I'll do that.

Reply:
Update: Dropped down to just an explicit mount.